tracer/src/Datadog.Trace/ClrProfiler/README.md

```diff
@@ -0,0 +1,119 @@
+3. Create an instrumentation class using one of the standard "shapes" (described below), and place it in the [ClrProfiler/AutoInstrumentation folder](./AutoInstrumentation). If the methods you need to instrument have different prototypes (especially the number of parameters), you will need multiple class to instrument them.
+4. Add an `[InstrumentMethod]` attribute to the instrumentation class, as described below. Alternatively, add an assembly-level `[AdoNetClientInstrumentMethods]` attribute
+5. (Optional) Create duck-typing unit tests in _Datadog.Trace.Tests_ to confirm any duck types are valid. This can make the feedback cycle much faster than relying on integration tests
+6. Create integration tests for your instrumentation 
+   1. Create (or reuse) a sample application that uses the target library, which ideally exercises all the code paths in your new instrumentation. Use an `$(ApiVersion)` MSBuild variables to allow testing against multiple package versions in CI. 
+   2. Add an entry in [tracer/build/PackageVersionsGeneratorDefinitions.json](../../../build/PackageVersionsGeneratorDefinitions.json) defining the range of all supported versions. See the existing definitions for examples
+   3. Run `./tracer/build.ps1 GeneratePackageVersions`. This generates the xunit test data for package versions in the `TestData` that you can use as `[MemberData]` for your `[Theory]` tests. 
+   4. Use the `MockTracerAgent` to confirm your instrumentation is working as expected.
+7. After testing locally, push to GitHub, and do a manual run in Azure Devops for your branch
+   1. Navigate to the [consolidated-pipeline](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build?definitionId=54)
+   2. Click `Run Pipeline`
+   3. Select your branch from the drop down
+   4. Click `Variables`, set `perform_comprehensive_testing` to true. (This is false for PRs by default for speed, but ensures your new code is tested against all the specified packages initially)
+   5. Select `Stages To Run`, and select only the `build*`, `unit_test*` and `integration_test*` stages. This avoids using excessive resources, and will complete your build faster
+8. Once your test branch works, create a PR!
+
+### Instrumentation classes
+
+When implementing instrumentation classes, you can run code both _before_ the target method is entered, and _after_ it is entered. Your `OnMethodBegin` method will always look the same, but the shape of the `OnMethodEnd` depends on whether the method is async, and whether it returns void:
+
+```csharp
+public class ClientQueryIteratorsIntegrations
+{
+    // The parameters here should match the method signature of the target method
+    // Use generic parameters for non-BCL types that you can't directly reference 
+    internal static CallTargetState OnMethodBegin<TTarget, TOther>(TTarget instance, TOther otherParam)
+    {
+        // Run your "method start" code here
+    }
+    
+    // Include ONE of the following:
+    
+    // 👇 Async method
```

Comment:
Perhaps we could clarify here in case there's any confusion regarding the vanilla Task type
```suggestion
    // 👇 Async method, with different behavior based on the return type
    //   - Task<TReturn>: returnValue is set to the Result
    //   - Task: TReturn is set to typeof(object) and returnValue is set to null
```

Reply:
Done in #2317